packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -3,50 +3,197 @@
-/// method to reparent your [FocusNode] if your widget moves from one
-/// location in the tree to another.
+/// [FocusNodes] are organized into "scopes" (see [FocusScopeNode]), which form
+/// sub-trees of nodes which can be traversed as a group. Within a scope, the
+/// most recent nodes to have focus are remembered, and if a node is focused and
+/// then removed, the original node receives focus again.
+///
+/// The focus node hierarchy can be traversed using the [parent], [children],
+/// [ancestors] and [descendants] accessors.
+///
+/// [FocusNode]s are [ChangeNotifier]s, so you can register to receive a
+/// notification when the focus changes. If you are using the [Focus] and
+/// [FocusScope] widgets to manage the nodes, it is often simpler to just
+/// establish an [InheritedWidget] dependency on them by calling [Focus.of] or
+/// [FocusScope.of] instead.
+///
+/// To see the focus tree in the debug console, you can call
+/// [debugDumpFocusTree].
 ///
 /// ## Lifetime
 ///
-/// Focus nodes are long-lived objects. For example, if a stateful widget has a
-/// focusable child widget, it should create a [FocusNode] in the
-/// [State.initState] method, and [dispose] it in the [State.dispose] method,
-/// providing the same [FocusNode] to the focusable child each time the
-/// [State.build] method is run. In particular, creating a [FocusNode] each time
-/// [State.build] is invoked will cause the focus to be lost each time the
-/// widget is built.
+/// Focus nodes are long-lived objects. For example, if a [StatefulWidget] can
+/// receive focus, and is not using a [Focus] widget, it could create a
+/// [FocusNode] in its [State.initState] method, and [dispose] it in the
+/// [State.dispose] method, providing the same [FocusNode] to the
+/// [reparentIfNeeded] call each time its [State.build] method is run. In
+/// particular, creating a [FocusNode] each time [State.build] is invoked will
+/// cause the focus to be lost each time the widget is built.
+///
+/// {@template flutter.widgets.focus_manager.focus.keyEvents}
+/// ## Key Event Propagation
+///
+/// The [FocusManager] receives all key events and will pass them to the focused
+/// nodes. It starts with the node with the primary focus, and will call the
+/// [onKey] callback for that node. If the callback returns false, indicating
+/// that it did not handle the event, the [FocusManager] will move to the parent
+/// of that node and call its [onKey]. If that [onKey] returns true, then it
+/// will stop propagating the event. If it reaches the root [FocusScopeNode],
+/// [FocusManager.rootScope], the event is discarded.
+/// {@endtemplate}
+///
+/// ## Focus Traversal
+///
+/// It is often that case that an application will want to give focus to the
+/// logical "next" or "previous" widget in the UI. Calling the [nextFocus],
+/// [previousFocus], and [focusInDirection] functions will perform these
+/// navigation operations on the focus tree.
```

Comment:
this can be written more tersely as:

```
To give focus to the logical "next" or "previous" widget in the UI, consider
calling the [nextFocus], [previousFocus], and [focusInDirection] methods.
```

Incidentally, functions -> methods unless they're global functions (not on a class).